query/aql_compiler.go

```diff
@@ -1786,12 +1786,28 @@ func (qc *AQLQueryContext) processMeasure() {
 	}
 }
 
+func (qc *AQLQueryContext) getAllColumns() (columns []expr.Expr) {
+	// only main table columns wildcard match supported
+	for _, column := range qc.TableScanners[0].Schema.Schema.Columns {
+		rawVarRef := &expr.VarRef{
+			Val: column.Name,
+		}
+		rewrittenVarRef := qc.Rewrite(rawVarRef)
+		columns = append(columns, rewrittenVarRef)
+	}
+	return
+}
+
 func (qc *AQLQueryContext) processDimensions() {
 	// Copy dimension ASTs.
-	qc.OOPK.Dimensions = make([]expr.Expr, len(qc.Query.Dimensions))
-	for i, dim := range qc.Query.Dimensions {
+	qc.OOPK.Dimensions = []expr.Expr{}
```

Comment:
how about after we detect it is a non-aggregation query and without dimensions set, then we just add columns into dimension list and then let the compiler automatically handle all other stuffs.

Reply:
I changed the implementation to expand wildcard at qc.Query.Dimension level. so processDimensions will remain unchanged.